test/load/locustfile-performance.py

```diff
@@ -10,66 +10,142 @@
-            # In this example, I've hardcoded response_length=0. If we would want the response length to be 
+            events.request_success.fire(request_type="grpc", name='send_tx_await', response_time=total_time, tx_hash=hex_hash, response_length=0)
+            # In this example, I've hardcoded response_length=0. If we would want the response length to be
             # reported correctly in the statistics, we would probably need to hook in at a lower level
 
+    def send_tx_await(self, transaction):
+        """
+        Send a transaction to Iroha and wait for the final status to be reported in status stream
+        :param transaction: protobuf Transaction
+        :return: None
+        """
+        while len(SENT) - len(COMMITTED) > 100:
+            time.sleep(0.01)
+
+        hex_hash = ascii_hash(transaction)
+        start_time = time.time()
+
+        try:
+            self.send_tx(transaction)
+            # tx_future = self._command_service_stub.Torii.future(transaction)
+            SENT.add(hex_hash)
+            TXS[hex_hash] = start_time
+        except grpc.RpcError as e:
+            total_time = int((time.time() - start_time) * 1000)
+            events.request_failure.fire(request_type="grpc", name='send_tx_await', response_time=total_time, tx_hash=hex_hash, exception=e)
+
+
+def block_listener(host):
+    iroha_api = iroha.Iroha("admin@test")
+    net = IrohaGrpc(host)
+    query = iroha_api.blocks_query()
+    ic.sign_query(query, ADMIN_PRIVATE_KEY)
+    print("Listeting blocks")
+    for block in net.send_blocks_stream_query(query):
+        BLOCKS.add(block.block_response.block.block_v1.payload.height)
+        hashes = block.block_response.block.block_v1.payload.rejected_transactions_hashes
+        txs =  block.block_response.block.block_v1.payload.transactions
+        for tx in txs:
+            hashes.append(ascii_hash(tx))
+
+        for hash in hashes:
+            if hash not in TXS.keys():
+                continue
+            start_time = TXS[hash]
+            COMMITTED.add(hash)
+            del TXS[hash]
+            total_time = int((time.time() - start_time) * 1000)
+            try:
+                events.request_success.fire(request_type="grpc", name='send_tx_await', response_time=total_time, tx_hash=hash, sent=start_time, committed=time.time(), response_length=0)
+            except Exception as e:
+                print(e)
 
 class IrohaLocust(Locust):
     """
     This is the abstract Locust class which should be subclassed. It provides an Iroha gRPC client
     that can be used to make gRPC requests that will be tracked in Locust's statistics.
     """
     def __init__(self, *args, **kwargs):
+        global LISTENING
         super(IrohaLocust, self).__init__(*args, **kwargs)
         self.client = IrohaClient(self.host)
+        if not LISTENING:
+            LISTENING = True
```

Comment:
Maybe just call `gevent.spawn` below the class definitions in order to avoid such a variable.